setup/vdisk.py

```diff
@@ -228,9 +222,9 @@ def create_clone(vdisk_name, vpool_name, new_vdisk_name, storagerouter_ip, api,
                                                                   snapshot_id))
             return task_result[1]
 
-    @staticmethod
+    @classmethod
     @required_vdisk
-    def set_vdisk_as_template(vdisk_name, vpool_name, api, timeout=SET_VDISK_AS_TEMPLATE_TIMEOUT):
+    def set_vdisk_as_template(vdisk_name, vpool_name, timeout=SET_VDISK_AS_TEMPLATE_TIMEOUT):
```

Comment:
Missing cls declaration. Snapshot_name is currently treated as the VDiskSetup pointer